lib/ansible/modules/network/cli/cli_command.py

```diff
@@ -0,0 +1,217 @@
+        then all conditionals in the I(wait_for) must be satisfied.  If
+        the value is set to C(any) then only one of the values must be
+        satisfied.
+    default: all
+    choices: ['any', 'all']
+    version_added: "2.2"
+  retries:
+    description:
+      - Specifies the number of retries a command should be tried
+        before it is considered failed.  The command is run on the
+        target device every retry and evaluated against the I(wait_for)
+        conditionals.
+    default: 10
+  interval:
+    description:
+      - Configures the interval in seconds to wait between retries
+        of the command.  If the command does not pass the specified
+        conditional, the interval indicates how to long to wait before
+        trying the command again.
+    default: 1
+"""
+
+EXAMPLES = """
+- name: run show version on remote devices
+  cli_command:
+    commands: show version
+
+- name: run show version and check to see if output contains Arista
+  cli_command:
+    commands: show version
+    wait_for: result[0] contains Arista
+
+- name: run multiple commands on remote nodes
+  cli_command:
+    commands:
+      - show version
+      - show interfaces
+
+- name: run multiple commands and evaluate the output
+  cli_command:
+    commands:
+      - show version
+      - show interfaces
+    wait_for:
+      - result[0] contains Arista
+      - result[1] contains Loopback0
+
+- name: run commands and specify the output format
+  cli_command:
+    commands:
+      - command: show version
+        output: json
+"""
+
+RETURN = """
+stdout:
+  description: The set of responses from the commands
+  returned: always apart from low level errors (such as action plugin)
+  type: list
+  sample: ['...', '...']
+stdout_lines:
+  description: The value of stdout split into a list
+  returned: always apart from low level errors (such as action plugin)
+  type: list
+  sample: [['...', '...'], ['...'], ['...']]
+failed_conditions:
+  description: The list of conditionals that have failed
+  returned: failed
+  type: list
+  sample: ['...', '...']
+"""
+import time
+
+from ansible.module_utils._text import to_native
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.connection import Connection
+from ansible.module_utils.network.common.parsing import Conditional
+from ansible.module_utils.network.common.utils import ComplexList, to_lines
+
+VALID_KEYS = ['command', 'output', 'prompt', 'response']
+
+
+def parse_commands(module, warnings):
+    transform = ComplexList(dict(
+        command=dict(key=True),
+        output=dict(),
+        prompt=dict(),
+        answer=dict()
+    ), module)
+
+    commands = transform(module.params['commands'])
+
+    if module.check_mode:
+        for item in list(commands):
+            if not item['command'].startswith('show'):
+                warnings.append(
+                    'Only show commands are supported when using check_mode, not '
```

Comment:
iirc we decided to allow configure commands as part of cli_command to support config commands that result in a command prompt

Reply:
Even for check mode?

Reply:
I missed the check_mode condition. Please ignore above comment